cli/tools/registry/pm.rs

```diff
@@ -0,0 +1,291 @@
+  let client = http_client.client()?;
+  let registry_api_url = jsr_api_url().to_string();
+
+  let mut selected_packages = Vec::with_capacity(add_flags.packages.len());
+  let mut package_reqs = Vec::with_capacity(add_flags.packages.len());
+
+  for package_name in add_flags.packages.iter() {
+    let req = if package_name.starts_with("npm:") {
+      let pkg_req = NpmPackageReqReference::from_str(package_name)
+        .with_context(|| {
+          format!("Failed to parse package required: {}", package_name)
+        })?;
+      AddPackageReq::Npm(pkg_req)
+    } else {
+      let pkg_req = JsrPackageReqReference::from_str(&format!(
+        "jsr:{}",
+        package_name.strip_prefix("jsr:").unwrap_or(package_name)
+      ))
+      .with_context(|| {
+        format!("Failed to parse package required: {}", package_name)
+      })?;
+      AddPackageReq::Jsr(pkg_req)
+    };
+
+    package_reqs.push(req);
+  }
+
+  let package_futures = package_reqs
+    .into_iter()
+    .map(|package_req| {
+      find_package_and_select_version_for_req(
+        client,
+        &registry_api_url,
+        package_req,
+      )
+      .boxed_local()
+    })
+    .collect::<Vec<_>>();
+
+  let stream_of_futures = deno_core::futures::stream::iter(package_futures);
+  let mut buffered = stream_of_futures.buffer_unordered(10);
+
+  while let Some(package_and_version_result) = buffered.next().await {
+    let package_and_version = package_and_version_result?;
+
+    match package_and_version {
+      PackageAndVersion::NotFound(package_name) => {
+        bail!("{} was not found.", crate::colors::red(package_name));
+      }
+      PackageAndVersion::Selected(selected) => {
+        selected_packages.push(selected);
+      }
+    }
+  }
+
+  let config_file_contents =
+    tokio::fs::read_to_string(&config_file_path).await.unwrap();
+  let ast = jsonc_parser::parse_to_ast(
+    &config_file_contents,
+    &Default::default(),
+    &Default::default(),
+  )?;
+
+  let obj = match ast.value {
+    Some(Value::Object(obj)) => obj,
+    _ => bail!("Failed updating config file due to no object."),
+  };
+
+  let mut existing_imports =
+    if let Some(imports) = config_file.json.imports.clone() {
+      match serde_json::from_value::<HashMap<String, String>>(imports) {
+        Ok(i) => i,
+        Err(_) => bail!("Malformed \"imports\" configuration"),
+      }
+    } else {
+      HashMap::default()
+    };
+
+  for selected_package in selected_packages {
+    eprintln!(
+      "Add {} - {}@{}",
+      crate::colors::green(&selected_package.import_name),
+      selected_package.package_name,
+      selected_package.version
+    );
+    existing_imports.insert(
+      selected_package.import_name,
+      format!(
+        "{}@{}",
+        selected_package.package_name, selected_package.version
+      ),
+    );
+  }
+  let mut import_list: Vec<(String, String)> =
+    existing_imports.into_iter().collect();
+
+  import_list.sort_by(|(k1, _), (k2, _)| k1.cmp(k2));
```

Comment:
I don't think we should sort the user's import list. Instead maybe we should attempt to insert the dependency sorted from bottom up (ex. start at the bottom and keep decrementing the insert index until the previous item is less than the current)? People might have their imports sorted in some semantical way.

Reply:
Ok, let's do this in a follow-up